src/Components/AffectedClustersTable/AffectedClustersTable.js

```diff
@@ -67,7 +67,9 @@ const AffectedClustersTable = ({ query, rule, afterDisableFn }) => {
   const page = filters.offset / filters.limit + 1;
   const allSelected =
     filteredRows.length !== 0 && selected.length === filteredRows.length;
-  const loadingState = isUninitialized || isFetching;
+  // helps to distinguish the state when the API data received but not yet filtered
+  const [rowsFiltered, setRowsFiltered] = useState(false);
```

Comment:
I would suggest renaming areRowsFiltered to explicitly mean it has a boolean type. On all other following boolean vars as well.

Reply:
But, I have seen that the application has a lot to rename if this is applied ^